File: DotNetGotchas/CSharp/ReflectionToSerialize/modified2/Serialization/Car.cs
//Car.cs
using System;
using System.Runtime.Serialization;

namespace Serialization
{
	[Serializable]
	public class Car : ISerializable	
	{
		private int yearOfMake;
		private Engine theEngine;
		private int miles = 0;

		public Car(int year, Engine anEngine)
		{
			yearOfMake = year;
			theEngine = anEngine;
		}

		public override string ToString()
		{
			return yearOfMake + ":" + miles + ":" + theEngine;
		}

		#region ISerializable Members

		public Car(SerializationInfo info, 
			StreamingContext context)
		{
			SerializationHelper.SetData(typeof(Car), this, info);
		}

		public virtual void GetObjectData(SerializationInfo info, 
			StreamingContext context)
		{
			SerializationHelper.GetData(typeof(Car), this, info);
		}

		#endregion
	}
}

File: DotNetGotchas/VBDN/ReflectionToSerialize/Modified2/Serialization/Car.vb
'Car.vb
Imports System.Runtime.Serialization
Imports System.Reflection

<Serializable()> _
Public Class Car
    Implements ISerializable

    Private yearOfMake As Integer
    Private theEngine As Engine
    Private miles As Integer = 0

    Public Sub New(ByVal year As Integer, ByVal anEngine As Engine)
        yearOfMake = year
        theEngine = anEngine
    End Sub

    Public Overrides Function ToString() As String
        Return yearOfMake & ":" & miles & ":" & theEngine.ToString()
    End Function

    Public Sub New( _
        ByVal info As SerializationInfo, _
        ByVal context As StreamingContext)

        SerializationHelper.SetData(GetType(Car), Me, info)
    End Sub

    Public Overridable Sub GetObjectData(ByVal info As SerializationInfo, _
        ByVal context As StreamingContext) _
            Implements ISerializable.GetObjectData

        SerializationHelper.GetData(GetType(Car), Me, info)
    End Sub
End Class
